cw_monero/lib/api/transaction_history.dart

```diff
@@ -1,138 +1,102 @@
-  return getTransactionNative(txIdPointer).ref;
+// TODO(mrcyjanek): ...
+Transaction getTransaction(String txId) {
+  return Transaction(txInfo: monero.TransactionHistory_transactionById(txhistory!, txid: txId));
 }
 
-PendingTransactionDescription createTransactionSync(
+Future<PendingTransactionDescription> createTransactionSync(
     {required String address,
     required String paymentId,
     required int priorityRaw,
     String? amount,
     int accountIndex = 0,
-    List<String> preferredInputs = const []}) {
-  final addressPointer = address.toNativeUtf8();
-  final paymentIdPointer = paymentId.toNativeUtf8();
-  final amountPointer = amount != null ? amount.toNativeUtf8() : nullptr;
-
-  final int preferredInputsSize = preferredInputs.length;
-  final List<Pointer<Utf8>> preferredInputsPointers =
-      preferredInputs.map((output) => output.toNativeUtf8()).toList();
-  final Pointer<Pointer<Utf8>> preferredInputsPointerPointer = calloc(preferredInputsSize);
-
-  for (int i = 0; i < preferredInputsSize; i++) {
-    preferredInputsPointerPointer[i] = preferredInputsPointers[i];
-  }
-
-  final errorMessagePointer = calloc<Utf8Box>();
-  final pendingTransactionRawPointer = calloc<PendingTransactionRaw>();
-  final created = transactionCreateNative(
-          addressPointer,
-          paymentIdPointer,
-          amountPointer,
-          priorityRaw,
-          accountIndex,
-          preferredInputsPointerPointer,
-          preferredInputsSize,
-          errorMessagePointer,
-          pendingTransactionRawPointer) !=
-      0;
-
-  calloc.free(preferredInputsPointerPointer);
-
-  preferredInputsPointers.forEach((element) => calloc.free(element));
-
-  calloc.free(addressPointer);
-  calloc.free(paymentIdPointer);
-
-  if (amountPointer != nullptr) {
-    calloc.free(amountPointer);
-  }
-
-  if (!created) {
-    final message = errorMessagePointer.ref.getValue();
-    calloc.free(errorMessagePointer);
+    List<String> preferredInputs = const []}) async {
+
+  final amt = amount == null ? 0 : monero.Wallet_amountFromString(amount);
+  
+  final address_ = address.toNativeUtf8(); 
+  final paymentId_ = paymentId.toNativeUtf8(); 
+  final preferredInputs_ = preferredInputs.join(monero.defaultSeparatorStr).toNativeUtf8();
+
+  final waddr = wptr!.address;
+  final addraddr = address_.address;
+  final paymentIdAddr = paymentId_.address;
+  final preferredInputsAddr = preferredInputs_.address;
+  final spaddr = monero.defaultSeparator.address;
+  final pendingTx = Pointer<Void>.fromAddress(await Isolate.run(() {
+    final tx = monero_gen.MoneroC(DynamicLibrary.open(monero.libPath)).MONERO_Wallet_createTransaction(
+      Pointer.fromAddress(waddr),
+      Pointer.fromAddress(addraddr).cast(),
+      Pointer.fromAddress(paymentIdAddr).cast(),
+      amt,
+      1,
+      priorityRaw,
+      accountIndex,
+      Pointer.fromAddress(preferredInputsAddr).cast(),
+      Pointer.fromAddress(spaddr),
+    );
```

Comment:
since monero_c/monero.dart runs everything on the main isolate, calling createTransaction would cause the UI to freeze, so we need to run it in an Isolate.run call. And because dart is especially bad at multi thread code, the only type that we can share between them is `int`, that's why we call .address on the pointers, turn them into pointers inside of the isolate, and free them later.